net/net.go

```diff
@@ -137,6 +137,30 @@ func IsIPv6CIDR(cidr *net.IPNet) bool {
 	return IsIPv6(ip)
 }
 
+// IsIPv4 returns if netIP is IPv4.
+func IsIPv4(netIP net.IP) bool {
+	return netIP != nil && netIP.To4() != nil
```

Comment:
This is the exact same code as `IsIPv6`, is there a problem somewhere?

Reply:
Yes just trying to condense the code that we have for dualstack here https://github.com/kubernetes/kubernetes/blob/master/test/e2e/network/dual_stack.go#L528-L542 .